File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookChapterDao.java
package ai.elimu.content_provider.room.dao;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ai.elimu.content_provider.room.entity.StoryBookChapter;

@Dao
public interface StoryBookChapterDao {

    @Insert
    void insert(StoryBookChapter storyBookChapter);

    @Query("SELECT * FROM StoryBookChapter WHERE id = :id")
    StoryBookChapter load(Long id);

    @Query("SELECT * FROM StoryBookChapter WHERE id = :id")
    Cursor loadAsCursor(Long id);

    @Query("SELECT * FROM StoryBookChapter WHERE storyBookId = :storyBookId")
    List<StoryBookChapter> loadAll(Long storyBookId);

    @Query("SELECT * FROM StoryBookChapter WHERE storyBookId = :storyBookId")
    Cursor loadAllAsCursor(Long storyBookId);

    @Update
    void update(StoryBookChapter storyBookChapter);

    @Query("DELETE FROM StoryBookChapter")
    void deleteAll();
}

File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookChapterDao.kt
package ai.elimu.content_provider.room.dao

import ai.elimu.content_provider.room.entity.StoryBookChapter
import android.database.Cursor
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface StoryBookChapterDao {
    @Insert
    fun insert(storyBookChapter: StoryBookChapter)

    @Query("SELECT * FROM StoryBookChapter WHERE id = :id")
    fun load(id: Long?): StoryBookChapter?

    @Query("SELECT * FROM StoryBookChapter WHERE id = :id")
    fun loadAsCursor(id: Long?): Cursor

    @Query("SELECT * FROM StoryBookChapter WHERE storyBookId = :storyBookId")
    fun loadAll(storyBookId: Long?): MutableList<StoryBookChapter>

    @Query("SELECT * FROM StoryBookChapter WHERE storyBookId = :storyBookId")
    fun loadAllAsCursor(storyBookId: Long?): Cursor

    @Update
    fun update(storyBookChapter: StoryBookChapter)

    @Query("DELETE FROM StoryBookChapter")
    fun deleteAll()
}
